smdebug/tensorflow/keras.py

```diff
@@ -805,6 +821,7 @@ def record_tensor_value(self, tensor_name, tensor_value):
             return
 
         self._add_metric(metric_name=tensor_name, metric_value=tensor_value)
+        self._write_optimizer_variables()
```

Comment:
all the changes in this PR is same as https://github.com/awslabs/sagemaker-debugger/pull/218/files , 
Why this extra change isr eqd, I mean why do we have to write optimizer variables at 2 different places , here and in L573

Reply:
We make a call to `self._write_optimizer_variables()` because the `on_batch_begin` and `on_batch_end` callbacks are not called with GradientTape.